### ISSUE TITLE
Error during install package

### ISSUE THREAD

**Issue by modche** [tsamsonov/grwat#8]
Hi,
the package is a nice effort for the community. As a hydrologist I am interested in different baseflow separation methods. I tried to install the package (after I have seen your talk at #useR2021). It fails... (see below), I have to install another package, I guess. However, there are a lot of warnings also... Btw, you have like 20+ dependencies, this is really a lot...

If you are interested, have a look on https://modche.github.io/delayedflow/.

Best + keep going!
Michael

```
> devtools::install_github("tsamsonov/grwat")
Downloading GitHub repo tsamsonov/grwat@HEAD
These packages have more recent versions available.
It is recommended to update all of them.
Which would you like to update?

1: All                         
2: CRAN packages only          
3: None                        
4: Rcpp (1.0.6 -> 1.0.7) [CRAN]
5: sf   (1.0-0 -> 1.0-1) [CRAN]

Enter one or more numbers, or an empty line to skip updates: 2
Rcpp        (1.0.6 -> 1.0.7   ) [CRAN]
R.methodsS3 (NA    -> 1.8.1   ) [CRAN]
lpSolve     (NA    -> 5.6.15  ) [CRAN]
sf          (1.0-0 -> 1.0-1   ) [CRAN]
tweenr      (NA    -> 1.0.2   ) [CRAN]
svglite     (NA    -> 2.0.0   ) [CRAN]
webshot     (NA    -> 0.5.2   ) [CRAN]
reshape     (NA    -> 0.8.8   ) [CRAN]
extraDistr  (NA    -> 1.9.1   ) [CRAN]
R.oo        (NA    -> 1.24.0  ) [CRAN]
transformr  (NA    -> 0.1.3   ) [CRAN]
gganimate   (NA    -> 1.0.7   ) [CRAN]
BiocManager (NA    -> 1.30.16 ) [CRAN]
kableExtra  (NA    -> 1.3.4   ) [CRAN]
GGally      (NA    -> 2.1.2   ) [CRAN]
mblm        (NA    -> 0.12.1  ) [CRAN]
trend       (NA    -> 1.1.4   ) [CRAN]
writexl     (NA    -> 1.4.0   ) [CRAN]
R.utils     (NA    -> 2.10.1  ) [CRAN]
Cairo       (NA    -> 1.5-12.2) [CRAN]
ncdf4       (NA    -> 1.17    ) [CRAN]
Installing 21 packages: Rcpp, R.methodsS3, lpSolve, sf, tweenr, svglite, webshot, reshape, extraDistr, R.oo, transformr, gganimate, BiocManager, kableExtra, GGally, mblm, trend, writexl, R.utils, Cairo, ncdf4

  There are binary versions available but the source versions are later:
     binary source needs_compilation
Rcpp  1.0.6  1.0.7              TRUE
sf    1.0-0  1.0-1              TRUE

Do you want to install from sources the packages which need compilation? (Yes/no/cancel) y
trying URL 'https://cran.rstudio.com/bin/macosx/big-sur-arm64/contrib/4.1/R.methodsS3_1.8.1.tgz'
Content type 'application/x-gzip' length 79811 bytes (77 KB)
==================================================
downloaded 77 KB

trying URL 'https://cran.rstudio.com/bin/macosx/big-sur-arm64/contrib/4.1/lpSolve_5.6.15.tgz'
Content type 'application/x-gzip' length 1076987 bytes (1.0 MB)
==================================================
downloaded 1.0 MB

trying URL 'https://cran.rstudio.com/bin/macosx/big-sur-arm64/contrib/4.1/tweenr_1.0.2.tgz'
Content type 'application/x-gzip' length 1142744 bytes (1.1 MB)
==================================================
downloaded 1.1 MB

trying URL 'https://cran.rstudio.com/bin/macosx/big-sur-arm64/contrib/4.1/svglite_2.0.0.tgz'
Content type 'application/x-gzip' length 872473 bytes (852 KB)
==================================================
downloaded 852 KB

trying URL 'https://cran.rstudio.com/bin/macosx/big-sur-arm64/contrib/4.1/webshot_0.5.2.tgz'
Content type 'application/x-gzip' length 133198 bytes (130 KB)
==================================================
downloaded 130 KB

trying URL 'https://cran.rstudio.com/bin/macosx/big-sur-arm64/contrib/4.1/reshape_0.8.8.tgz'
Content type 'application/x-gzip' length 168277 bytes (164 KB)
==================================================
downloaded 164 KB

trying URL 'https://cran.rstudio.com/bin/macosx/big-sur-arm64/contrib/4.1/extraDistr_1.9.1.tgz'
Content type 'application/x-gzip' length 3856402 bytes (3.7 MB)
==================================================
downloaded 3.7 MB

trying URL 'https://cran.rstudio.com/bin/macosx/big-sur-arm64/contrib/4.1/R.oo_1.24.0.tgz'
Content type 'application/x-gzip' length 965316 bytes (942 KB)
==================================================
downloaded 942 KB

trying URL 'https://cran.rstudio.com/bin/macosx/big-sur-arm64/contrib/4.1/transformr_0.1.3.tgz'
Content type 'application/x-gzip' length 630870 bytes (616 KB)
==================================================
downloaded 616 KB

trying URL 'https://cran.rstudio.com/bin/macosx/big-sur-arm64/contrib/4.1/gganimate_1.0.7.tgz'
Content type 'application/x-gzip' length 1599691 bytes (1.5 MB)
==================================================
downloaded 1.5 MB

trying URL 'https://cran.rstudio.com/bin/macosx/big-sur-arm64/contrib/4.1/BiocManager_1.30.16.tgz'
Content type 'application/x-gzip' length 326490 bytes (318 KB)
==================================================
downloaded 318 KB

trying URL 'https://cran.rstudio.com/bin/macosx/big-sur-arm64/contrib/4.1/kableExtra_1.3.4.tgz'
Content type 'application/x-gzip' length 1848978 bytes (1.8 MB)
==================================================
downloaded 1.8 MB

trying URL 'https://cran.rstudio.com/bin/macosx/big-sur-arm64/contrib/4.1/GGally_2.1.2.tgz'
Content type 'application/x-gzip' length 1636398 bytes (1.6 MB)
==================================================
downloaded 1.6 MB

trying URL 'https://cran.rstudio.com/bin/macosx/big-sur-arm64/contrib/4.1/mblm_0.12.1.tgz'
Content type 'application/x-gzip' length 24719 bytes (24 KB)
==================================================
downloaded 24 KB

trying URL 'https://cran.rstudio.com/bin/macosx/big-sur-arm64/contrib/4.1/trend_1.1.4.tgz'
Content type 'application/x-gzip' length 388875 bytes (379 KB)
==================================================
downloaded 379 KB

trying URL 'https://cran.rstudio.com/bin/macosx/big-sur-arm64/contrib/4.1/writexl_1.4.0.tgz'
Content type 'application/x-gzip' length 477226 bytes (466 KB)
==================================================
downloaded 466 KB

trying URL 'https://cran.rstudio.com/bin/macosx/big-sur-arm64/contrib/4.1/R.utils_2.10.1.tgz'
Content type 'application/x-gzip' length 1406328 bytes (1.3 MB)
==================================================
downloaded 1.3 MB

trying URL 'https://cran.rstudio.com/bin/macosx/big-sur-arm64/contrib/4.1/Cairo_1.5-12.2.tgz'
Content type 'application/x-gzip' length 9957634 bytes (9.5 MB)
==================================================
downloaded 9.5 MB

trying URL 'https://cran.rstudio.com/bin/macosx/big-sur-arm64/contrib/4.1/ncdf4_1.17.tgz'
Content type 'application/x-gzip' length 2238420 bytes (2.1 MB)
==================================================
downloaded 2.1 MB


The downloaded binary packages are in
	/var/folders/vx/5k__vfj97817p74wx5qg0vr00000gn/T//RtmpguFg4a/downloaded_packages
installing the source packages ‘Rcpp’, ‘sf’

trying URL 'https://cran.rstudio.com/src/contrib/Rcpp_1.0.7.tar.gz'
Content type 'application/x-gzip' length 2941808 bytes (2.8 MB)
==================================================
downloaded 2.8 MB

trying URL 'https://cran.rstudio.com/src/contrib/sf_1.0-1.tar.gz'
Content type 'application/x-gzip' length 3644763 bytes (3.5 MB)
==================================================
downloaded 3.5 MB

* installing *source* package ‘Rcpp’ ...
** package ‘Rcpp’ successfully unpacked and MD5 sums checked
** using staged installation
** libs
clang++ -arch arm64 -std=gnu++14 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I../inst/include/  -I/opt/R/arm64/include   -fPIC  -falign-functions=64 -Wall -g -O2  -c api.cpp -o api.o
clang++ -arch arm64 -std=gnu++14 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I../inst/include/  -I/opt/R/arm64/include   -fPIC  -falign-functions=64 -Wall -g -O2  -c attributes.cpp -o attributes.o
clang++ -arch arm64 -std=gnu++14 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I../inst/include/  -I/opt/R/arm64/include   -fPIC  -falign-functions=64 -Wall -g -O2  -c barrier.cpp -o barrier.o
clang++ -arch arm64 -std=gnu++14 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I../inst/include/  -I/opt/R/arm64/include   -fPIC  -falign-functions=64 -Wall -g -O2  -c date.cpp -o date.o
clang++ -arch arm64 -std=gnu++14 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I../inst/include/  -I/opt/R/arm64/include   -fPIC  -falign-functions=64 -Wall -g -O2  -c module.cpp -o module.o
clang++ -arch arm64 -std=gnu++14 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I../inst/include/  -I/opt/R/arm64/include   -fPIC  -falign-functions=64 -Wall -g -O2  -c rcpp_init.cpp -o rcpp_init.o
clang++ -arch arm64 -std=gnu++14 -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -single_module -multiply_defined suppress -L/Library/Frameworks/R.framework/Resources/lib -L/opt/R/arm64/lib -o Rcpp.so api.o attributes.o barrier.o date.o module.o rcpp_init.o -F/Library/Frameworks/R.framework/.. -framework R -Wl,-framework -Wl,CoreFoundation
installing to /Library/Frameworks/R.framework/Versions/4.1-arm64/Resources/library/00LOCK-Rcpp/00new/Rcpp/libs
** R
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded from temporary location
** checking absolute paths in shared objects and dynamic libraries
** testing if installed package can be loaded from final location
** testing if installed package keeps a record of temporary installation path
* DONE (Rcpp)
* installing *source* package ‘sf’ ...
** package ‘sf’ successfully unpacked and MD5 sums checked
** using staged installation
configure: CC: clang -arch arm64
configure: CXX: clang++ -arch arm64 -std=gnu++11
checking for gdal-config... no
no
configure: error: gdal-config not found or not executable.
ERROR: configuration failed for package ‘sf’
* removing ‘/Library/Frameworks/R.framework/Versions/4.1-arm64/Resources/library/sf’
* restoring previous ‘/Library/Frameworks/R.framework/Versions/4.1-arm64/Resources/library/sf’

The downloaded source packages are in
	‘/private/var/folders/vx/5k__vfj97817p74wx5qg0vr00000gn/T/RtmpguFg4a/downloaded_packages’
✓  checking for file ‘/private/var/folders/vx/5k__vfj97817p74wx5qg0vr00000gn/T/RtmpguFg4a/remotes52e37f0ed4ef/tsamsonov-grwat-e06460f/DESCRIPTION’ ...
─  preparing ‘grwat’:
✓  checking DESCRIPTION meta-information ...
─  cleaning src
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
   Removed empty directory ‘grwat/slides/grwat_dev’
─  looking to see if a ‘data/datalist’ file should be added
─  building ‘grwat_0.0.1.tar.gz’
   
* installing *source* package ‘grwat’ ...
** using staged installation
** libs
clang++ -arch arm64 -std=gnu++17 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG  -I'/Library/Frameworks/R.framework/Versions/4.1-arm64/Resources/library/Rcpp/include' -I/opt/R/arm64/include   -fPIC  -falign-functions=64 -Wall -g -O2  -c RcppExports.cpp -o RcppExports.o
clang++ -arch arm64 -std=gnu++17 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG  -I'/Library/Frameworks/R.framework/Versions/4.1-arm64/Resources/library/Rcpp/include' -I/opt/R/arm64/include   -fPIC  -falign-functions=64 -Wall -g -O2  -c grwat_core.cpp -o grwat_core.o
grwat_core.cpp:411:23: warning: unused variable 'g' [-Wunused-variable]
            for (auto g: FactGapsin) {
                      ^
grwat_core.cpp:464:18: warning: unused variable 'Qmax' [-Wunused-variable]
            auto Qmax = Qin[nmax];
                 ^
grwat_core.cpp:104:27: warning: unused function 'get_baseflow_singlepass' [-Wunused-function]
    static vector<double> get_baseflow_singlepass(const vector<double>& Qin,
                          ^
grwat_core.cpp:134:27: warning: unused function 'get_baseflow_recursive' [-Wunused-function]
    static vector<double> get_baseflow_recursive(const vector<double>& Qin,
                          ^
grwat_core.cpp:247:17: warning: unused function 'separate' [-Wunused-function]
    static void separate(const vector<int>& Year, const vector<int>& Mon, const vector<int>& Day,
                ^
5 warnings generated.
clang++ -arch arm64 -std=gnu++17 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG  -I'/Library/Frameworks/R.framework/Versions/4.1-arm64/Resources/library/Rcpp/include' -I/opt/R/arm64/include   -fPIC  -falign-functions=64 -Wall -g -O2  -c separate.cpp -o separate.o
In file included from separate.cpp:1:
./grwat_core.cpp:411:23: warning: unused variable 'g' [-Wunused-variable]
            for (auto g: FactGapsin) {
                      ^
./grwat_core.cpp:464:18: warning: unused variable 'Qmax' [-Wunused-variable]
            auto Qmax = Qin[nmax];
                 ^
separate.cpp:60:11: warning: enumeration value 'FUREY' not handled in switch [-Wswitch]
  switch (b) 
          ^
separate.cpp:60:11: note: add missing switch cases
  switch (b) 
          ^
separate.cpp:70:1: warning: non-void function does not return a value in all control paths [-Wreturn-type]
}
^
4 warnings generated.
clang++ -arch arm64 -std=gnu++17 -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -single_module -multiply_defined suppress -L/Library/Frameworks/R.framework/Resources/lib -L/opt/R/arm64/lib -o grwat.so RcppExports.o grwat_core.o separate.o -F/Library/Frameworks/R.framework/.. -framework R -Wl,-framework -Wl,CoreFoundation
installing to /Library/Frameworks/R.framework/Versions/4.1-arm64/Resources/library/00LOCK-grwat/00new/grwat/libs
** R
** inst
** byte-compile and prepare package for lazy loading
Warning in knitr::write_bib(c(.packages(), "Scale4C"), "./vignettes/packages.bib") :
  package(s) Scale4C not found
✓ Setting active project to '/private/var/folders/vx/5k__vfj97817p74wx5qg0vr00000gn/T/RtmpIMyP4X/R.INSTALL65c93e16d7f4/grwat'
✓ Adding 'Rcpp' to Imports field in DESCRIPTION
• Refer to functions with `Rcpp::fun()`
• Refer to functions with `sf::fun()`
• Refer to functions with `zoo::fun()`
• Refer to functions with `shiny::fun()`
• Refer to functions with `grid::fun()`
• Refer to functions with `ncdf4::fun()`
• Refer to functions with `lubridate::fun()`
• Refer to functions with `dplyr::fun()`
Warning in if (delta < 0) { :
  the condition has length > 1 and only the first element will be used
Warning in if (delta > 0) { :
  the condition has length > 1 and only the first element will be used
• Refer to functions with `rmarkdown::fun()`
Error : The `Cairo` package is required.
Error: unable to load R code in package ‘grwat’
Execution halted
ERROR: lazy loading failed for package ‘grwat’
* removing ‘/Library/Frameworks/R.framework/Versions/4.1-arm64/Resources/library/grwat’
Warning messages:
1: In i.p(...) : installation of package ‘sf’ had non-zero exit status
2: In i.p(...) :
  installation of package ‘/var/folders/vx/5k__vfj97817p74wx5qg0vr00000gn/T//RtmpguFg4a/file52e3769ad33b/grwat_0.0.1.tar.gz’ had non-zero exit status`
```

**Comment by tsamsonov** [comment]
Hi @modche ! Thanks for your interest and quick feedback. I need a major overhaul of the package to reduce its dependency and make its installation more stable. Up to now I was mainly interested in growing its functionality. I will get back here as soon as the new version is available.

And thanks for your link and paper! Very interesting, I will share it with my colleagues.

**Comment by tsamsonov** [comment]
@modche Do you use ARM-based Mac computer?

**Comment by modche** [comment]
Yes, Mac Book Air M1 2020

**Comment by tsamsonov** [comment]
@modche the package is now released on CRAN under version 0.0.2. Please check if you can install without problems.
